### PR TITLE
feat(simulator): adds changePubKey sim

### PR DESCRIPTION
**Pull request by jurajpiar**
- adds changePubKey simulations

Task: [ROLLUP-245](https://rsklabs.atlassian.net/browse/ROLLUP-245)

### CODE REVIEW

File: infrastructure/simulation-tool/src/simulations/changePubKey.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+                token: constants.AddressZero,
+                amount: expectedFee,
+                to: from.address()
+            });
+            await tx.awaitReceipt();
+        }
+    }
+
+    // execute pubKeyChanges
+    const executedTx = await Promise.all((await executePubKeyChanges(preparedPubKeyChanges, txDelay)).map((tx) => tx));
```
Comment (antomor) [inline infrastructure/simulation-tool/src/simulations/changePubKey.ts:35]: I guess we don't really need `map((tx) => tx)`
```suggestion
    const executedTx = await Promise.all((await executePubKeyChanges(preparedPubKeyChanges, txDelay)));
```

File: infrastructure/simulation-tool/test/operations/changePubKey.test.ts [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+            const { from } = preparePubKeyChange(expectedSender);
+
+            expect(from).to.deep.equal(expectedSender);
+        });
+
+        it('should return object with "ethAuthType" set to "ECDSA"', () => {
+            const expectedValue = 'ECDSA';
+            const { ethAuthType } = preparePubKeyChange(senderWallet);
+
+            expect(ethAuthType).to.deep.equal(expectedValue);
```
Comment (antomor) [inline infrastructure/simulation-tool/test/operations/changePubKey.test.ts:41]: I know it doesn't hurt, but we don't probably need `deep` here

File: infrastructure/simulation-tool/test/operations/changePubKey.test.ts [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+        });
+
+        it('should return object with "feeToken" set to zero address', () => {
+            const expectedFeeToken = constants.AddressZero;
+            const { feeToken } = preparePubKeyChange(senderWallet);
+
+            expect(feeToken).to.deep.equal(expectedFeeToken);
```
Comment (antomor) [inline infrastructure/simulation-tool/test/operations/changePubKey.test.ts:48]: I know it doesn't hurt, but we don't probably need deep here

File: infrastructure/simulation-tool/test/operations/changePubKey.test.ts [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+
+                return stub;
+            };
+
+            const walletGenerator: RollupWalletGenerator = {
+                next: sinon.stub().callsFake(() => Promise.resolve({ value: createWalletStub(genCount++) }))
+            } as unknown as RollupWalletGenerator;
+
+            const results = await generatePubKeyChanges(txCount, walletGenerator);
+            const actualResults = await Promise.all(results.map(async ({ from }) => await from.isSigningKeySet()));
```
Comment (antomor) [inline infrastructure/simulation-tool/test/operations/changePubKey.test.ts:83]: Why do we need both `await Promise.all...` and `await from.isSigningKeySet`?

File: infrastructure/simulation-tool/test/operations/changePubKey.test.ts [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+                { from: senderWallet, ethAuthType: 'ECDSA', feeToken: constants.AddressZero },
+                { from: senderWallet, ethAuthType: 'ECDSA', feeToken: constants.AddressZero }
+            ];
+
+            const actualResults = await executePubKeyChanges(preparedPubKeyChanges, 0);
+
+            const allResultsAreTransactions = actualResults.reduce((acc, next) => {
+                acc = acc && next instanceof Promise;
+                return acc;
+            }, true);
```
Comment (antomor) [inline infrastructure/simulation-tool/test/operations/changePubKey.test.ts:152]: Is there any reason for not to use [Array.every](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every)?

File: infrastructure/simulation-tool/test/operations/changePubKey.test.ts [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+            ];
+
+            const actualResults = await Promise.all(
+                (await executePubKeyChanges(preparedPubKeyChanges, 0)).map(async (tx) => await tx)
+            );
+
+            const allResultsAreTransactions = actualResults.reduce((acc, next) => {
+                acc = acc && next instanceof Transaction;
+                return acc;
+            }, true);
```
Comment (antomor) [inline infrastructure/simulation-tool/test/operations/changePubKey.test.ts:173]: Unless there is a specific reason, I'd use [Array.every](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every)

**Review by antomor — DISMISSED** [review]
@jurajpiar I've just left some comments and most of them are related to the tests.

File: infrastructure/simulation-tool/src/simulations/changePubKey.ts (hunk 1 above)
Comment (jurajpiar) [reply infrastructure/simulation-tool/src/simulations/changePubKey.ts]: done

File: infrastructure/simulation-tool/test/operations/changePubKey.test.ts (hunk 2 above)
Comment (jurajpiar) [reply infrastructure/simulation-tool/test/operations/changePubKey.test.ts]: done

File: infrastructure/simulation-tool/test/operations/changePubKey.test.ts (hunk 3 above)
Comment (jurajpiar) [reply infrastructure/simulation-tool/test/operations/changePubKey.test.ts]: done

File: infrastructure/simulation-tool/test/operations/changePubKey.test.ts (hunk 4 above)
Comment (jurajpiar) [reply infrastructure/simulation-tool/test/operations/changePubKey.test.ts]: done

File: infrastructure/simulation-tool/test/operations/changePubKey.test.ts (hunk 5 above)
Comment (jurajpiar) [reply infrastructure/simulation-tool/test/operations/changePubKey.test.ts]: done

File: infrastructure/simulation-tool/test/operations/changePubKey.test.ts (hunk 6 above)
Comment (jurajpiar) [reply infrastructure/simulation-tool/test/operations/changePubKey.test.ts]: done

**Review by antomor — APPROVED** [review]
LGTM

**Pull request merged**